docs/Analyzer reference page template.md

```diff
@@ -0,0 +1,31 @@
+# RULEID: Friendly rule name
+
+## Cause
+
+## Rule description
+
+## How to fix violations
+
+## When to suppress warnings
+
+## Example of a violation
```

Comment:
Actually, we ditched "issue" in favor of "result" to allow for log entries that were purely informational, or that reported a "pass" ("I evaluated rule X against target Y and all is well.").

Now, the spec defines a "result" as a "condition present in a programming artifact", and it defines a "problem" as a "result which indicates a condition that has the potential to detract from the quality of the program". So it would be entirely within the spirit and letter of SARIF to change those headings to "How to fix problems"  and "Example of a problem".

I'd rather not do that, for this reason: in the context of the SARIF spec, "problem" has a very specific, non-judgmental meaning: it's something that has the _potential_ to detract from quality. But in the context of non-SARIF experts (which is everybody!) reading a reference page, "problem" is a subjective word. "Violation" is objective: the rule said to do X, and you did non-X instead, so you unquestionably violated the rule. But it's up to the user to decide whether that violation constitutes a problem.